proposed/error-result.md

```diff
@@ -0,0 +1,234 @@
+# DRAFT: Error and Result Handling
+
+## 1. Overview
+
+This document describes standard interfaces for representing operation results in PHP applications.
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in [RFC 2119](https://tools.ietf.org/html/rfc2119).
+
+## 2. Definitions
+
+- **Result**: The outcome of an operation that may succeed or fail.
+- **Success Result**: A result containing a succesful value.
+- **Failure Result**: A result containing error information.
+- **Error**: Structured information about why an operation failed.
+
+## 3. Interfaces
+
+### 3.1 `ErrorInterface`
+
+```php
+<?php
+
+namespace Psr\Error;
```

Comment:
The namespace for ErrorInterface and ResultInterface should be the same, in my opinion:


```suggestion
namespace Psr\Result;
```